### ISSUE TITLE
Grouping exercises within workout

### ISSUE THREAD

**Issue by rolandgeider** [wger-project/wger#142]
Idea: there could be 'slots' of exercises for warmup, stretching, the workout itself etc.

**Comment by igoooor** [comment]
Yes this would be to fit routines you can find like in here http://www.reddit.com/r/bodyweightfitness/wiki/kb/recommended_routine
You can see that you have groups of exercises within the workout. It would just be a way to "organize" your workout :)

**Comment by rolandgeider** [comment]
Also the slots should not have an intrinsic meaning (i.e. just a label), so that users could define them on their own (warmup and workout should still be the default) if they do something else.

**Comment by alelevinas** [comment]
Ok, this seems like something i could do this week.

So i was thinking to create a new Model class like ExerciseCategory or Muscle and add it as a Foreign Key to Exercise. Then i'll think about how to present it. (maybe similar to the Exercise overview with tabs and the possibility to add a new slot)

**Comment by alelevinas** [comment]
As Category is taken and Slot is a little confusing i'm having trouble how we should name this "part of the workout".

Examples of these "part of the workout" could be

- Warmup

- Skill work

- Strengh work

- The Cool Down

- Aerobic Training

- Balance

etc.

As they would also be alble to define by users maybe Rumba exersices haha.

**Comment by rolandgeider** [comment]
@alelevinas you're on fire :)

Since the goal for this is to group the exercises within a single workout, basically just to make it visually nicer, the foreign key would need to go from Set to Workout. Actually, if we really wanted to make it correctly, we'd have to do it like this:

```
workout -> day -> slot -> set -> setting
```

However, this would break a lot of things and I'm not sure we'd gain much from it apart from increasing the complexity of the data model. What do you think about just adding a new text field to the set table and sort and group by that? We can use a Slot table where we store the default values for the names as well as the user-defined ones.

**Comment by alelevinas** [comment]
ahhhhh, i thought it was an overall category for every exercise.
So this is to organize your workout per day per part of the day-workout.

Cool, let's do that.

**Comment by rolandgeider** [comment]
Yes, this is just for information purposes and to mark the different exercises (after all, on one workout an exercise can be just for warm up but "for real" on another). I would think that "slots" is a good name for this.

I was thinking that a better modelling than just the text field from my other message would be to use a foreign key from set to a new and simple Slots table:

```
Slot
====
id
name
user --> foreign key to user or NULL
```

Entries without a user foreign key would be "system" entries (to keep it simple, I'd just add "Warmup", "Workout" and "Cool down" in the beginning). Users can add their own, and when they select one option they only see their own and the system wide ones. Just some thoughts.

**Comment by alelevinas** [comment]
I like that. So we'd ve having more than one set for each day, i presume?

I mean,
For Day A: Set 1 associated to Slot "Warm up" and Day A
                  Set 2 associated to Slot "Workout" and Day A
                  etc.

Just to be sure cause you have the entire model in your head and i've read it just a couple of week and sometimes i get mixed up.

**Comment by rolandgeider** [comment]
Yeah, exactly like that

**Comment by rolandgeider** [comment]
Hi @alelevinas! Did you start with this? If you get stuck or something, just ping me or something :)
